### PR TITLE
104081 publish integration events

### PR DESCRIPTION
**Pull request by janid1967**
[AB#104081](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/104081)

### CODE REVIEW

File: src/tests/Equinor.ProCoSys.Completion.Infrastructure.Tests/Repositories/PunchItemRepositoryTests.cs [hunk 1]
```diff
@@ -54,17 +54,30 @@ protected override void SetupRepositoryWithOneKnownItem()
...
-        Assert.AreEqual(_raisedByOrg, result.RaisedByOrg);
+
+        Assert.IsNotNull(result.Project);
+        Assert.AreEqual(_project.Guid, result.Project.Guid);
+
+        Assert.IsNotNull(result.RaisedByOrg);
+        Assert.AreEqual(_raisedByOrg.Guid, result.RaisedByOrg.Guid);
+        
+        Assert.IsNotNull(result.ClearingByOrg);
         Assert.AreEqual(_clearingByOrg, result.ClearingByOrg);
```
Comment (jsoi-eq) [inline src/tests/Equinor.ProCoSys.Completion.Infrastructure.Tests/Repositories/PunchItemRepositoryTests.cs:72]: Curious: Why is this one not asserting on Guid property like the other asserts?

Comment (janid1967) [reply src/tests/Equinor.ProCoSys.Completion.Infrastructure.Tests/Repositories/PunchItemRepositoryTests.cs:72]: Should assert on Guids as other ones

File: src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/UpdatePunchItem/UpdatePunchItemCommandHandler.cs [hunk 2]
```diff
@@ -41,148 +45,156 @@ public async Task<Result<string>> Handle(UpdatePunchItemCommand request, Cancell
...
+        PunchItem punchItem,
+        JsonPatchDocument<PatchablePunchItem> patchDocument)
     {
-        var operations = request.PatchDocument.Operations;
+        var changes = new List<IProperty>();
 
-        var replaceProperties = operations.Select(op => op.path.TrimStart('/')).ToList();
-        if (!replaceProperties.Any())
+        var propertiesToReplace = GetPropertiesToReplace(patchDocument);
+      if (!propertiesToReplace.Any())
```
Comment (jsoi-eq) [inline src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/UpdatePunchItem/UpdatePunchItemCommandHandler.cs:70]: Indent line

File: src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/UpdatePunchItem/UpdatePunchItemCommandHandler.cs [hunk 3]
```diff
@@ -41,148 +45,156 @@ public async Task<Result<string>> Handle(UpdatePunchItemCommand request, Cancell
...
-        foreach (var prop in replaceProperties)
+        foreach (var propertyToReplace in propertiesToReplace)
         {
-            switch (prop)
+            switch (propertyToReplace)
             {
                 case nameof(PatchablePunchItem.Description):
-                    punchItem.Description = patchedPunchItem.Description;
-                    patch.Description = patchedPunchItem.Description;
+                    if (punchItem.Description != patchedPunchItem.Description)
```
Comment (jsoi-eq) [inline src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/UpdatePunchItem/UpdatePunchItemCommandHandler.cs:84]: Extract implementation details to methods?
                case nameof(PatchablePunchItem.Description):
                    SetDescription(punchItem, patchedPunchItem, changes);
                    break;

File: src/tests/Equinor.ProCoSys.Completion.Command.Tests/PunchItemCommands/UpdatePunchItem/UpdatePunchItemCommandHandlerTests.cs [hunk 4]
```diff
@@ -133,8 +135,112 @@ public async Task HandlingCommand_ShouldAddPunchItemUpdatedEvent_WhenOperationsG
...
+            punchItemUpdatedDomainEventAdded
+                .Changes
+                .SingleOrDefault(c => c.Name == nameof(PunchItem.Type)),
+            oldTypeCode,
+            null);
+    }
+
     #endregion
 
     #region test update when patchDocument don't contains any operations
```
Comment (jsoi-eq) [inline src/tests/Equinor.ProCoSys.Completion.Command.Tests/PunchItemCommands/UpdatePunchItem/UpdatePunchItemCommandHandlerTests.cs:246]: We only test Description being the same leads to no update.
I think we ought to test for the other properties too, to test the if-block for each property in the switch-case. 
E.g. if (punchItem.RaisedByOrg.Guid != patchedPunchItem.RaisedByOrgGuid)

**Review by jsoi-eq — COMMENTED** [review]
Awaiting feedback.

File: src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/UpdatePunchItem/UpdatePunchItemCommandHandler.cs (hunk 3 above)
Comment (janid1967) [reply src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/UpdatePunchItem/UpdatePunchItemCommandHandler.cs:84]: Fine for me. Agree @TordJoranger ?

File: src/tests/Equinor.ProCoSys.Completion.Command.Tests/PunchItemCommands/UpdatePunchItem/UpdatePunchItemCommandHandlerTests.cs (hunk 4 above)
Comment (janid1967) [reply src/tests/Equinor.ProCoSys.Completion.Command.Tests/PunchItemCommands/UpdatePunchItem/UpdatePunchItemCommandHandlerTests.cs:246]: I'll fix

File: src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/UpdatePunchItem/UpdatePunchItemCommandHandler.cs (hunk 3 above)
Comment (TordJoranger) [reply src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/UpdatePunchItem/UpdatePunchItemCommandHandler.cs:84]: I'm not sure, but its not my hill to die on.

If you both agree its better, I guess i do too. But lets call it Update or Patch Description.

Comment (janid1967) [reply src/Equinor.ProCoSys.Completion.Command/PunchItemCommands/UpdatePunchItem/UpdatePunchItemCommandHandler.cs]: image

**Review by jsoi-eq — APPROVED** [review]

**Pull request merged**